src/Npgsql/NpgsqlConnection.cs

```diff
@@ -1431,6 +1437,20 @@ public void ReloadTypes()
         }
 
         #endregion Misc
+
+        /// <summary>
+        /// Collects statistics for the connection/session
+        /// </summary>
+        [PublicAPI]
+        public IDictionary<string, long> RetrieveStatistics()
+        {
+            if (EnableStatistics && Connector != null)
+            {
+                return Connector.RetrieveStatistics();
+            }
+            
+            return new Dictionary<string, long>();
```

Comment:
Could you return a reference to a static instance of an empty `ReadOnlyDictionary<string, long>`?